adafruit_sharpmemorydisplay.py

```diff
@@ -27,13 +27,16 @@
 
 """
 # pylint: enable=line-too-long
+from __future__ import annotations
```

Comment:
For the imports we want to group them all together inside of their own try block (seperate from the numpy import try block that exists in this file). The first import in the block should be from `typing` so it will raise it's importerror on microcontrollers and prevent the importing of any of the modules needed for types only. Which saves RAM on the microcontroller during execution.

The try block should look something like this:
```
try:
    import typing
    from __future__ import annotations
    from busio import SPI
    from digitalio import DigitalInOut
    from circuitpython_typing.pil import Image
except ImportError:
    pass
```
Any imports that are used only in typing are included in that block.

Reply:
@FoamyGuy Mypy was complaining that there was no use of `typing` so it didn't make it into my resolved code.

Reply:
CI is failing due to 
`from` __future__ import annotations
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
SyntaxError: from __future__ imports must occur at the beginning of the file` 

not being the first line after the docstring. Moving it back to the first item after the docstring.

Reply:
We need to have the "unused" `typing` import in the file. In our case when the code gets run on the microcontroller that "unused" import actually does have a specific purpose of triggering the ImportError which causes the remaining imports needed for typing only not to get imported which saves RAM when the code runs in microcontroller environment.

In cases where the import isn't used for anything we can add the pylint disable flag to make CI happy. If mypy also has a warning about this we'll need to figure out how to disable that eventually because we'll want to keep that "unused" import unless or until we have some other mechanism that prevents importing of things not strictly needed during runtime.